tidy3d/components/data/dataset.py

```diff
@@ -198,6 +199,32 @@ def symmetry_eigenvalues(self) -> Dict[str, Callable[[Axis], float]]:
             Hz=lambda dim: +1 if (dim == 2) else -1,
         )
 
+    def _restrict_to_box(self, box: Box):
```

Comment:
in the backend... I'll move it there...

Reply:
it's used for `EMEModeSolverMonitor`. Currently, the behavior is to take the EME modes (which span the entire simulation in the tangential directions) and restrict them to the `EMEModeSolverMonitor` geometry.

Reply:
The EMEModeSolverData has a list of ModeSolverData, each of which is associated to a ModeSolverMonitor. These monitors aren't in the list of simulation monitors, but rather they are just used internally while computing modes. But they are tied to the ModeSolverData that is returned.

Reply:
I think I'll change this behavior though, to just store the full `ModeSolverData` on any mode plane intersecting the monitor geometry.

Reply:
because otherwise I'd probably want to restrict the `ModeSolverMonitor` as well, but those are required to be 2D and could often become 1D after restriction